HIF.md

```diff
@@ -0,0 +1,810 @@
+|`EINVAL_KEY_INDEX`    |`0x100b`| Invalid key index.                                          |
+|`EINVAL_FRAME_LEN`    |`0x100c`| Invalid IEEE 802.15.4 frame length.                         |
+|`EINVAL_FRAME_VERSION`|`0x100d`| Invalid IEEE 802.15.4 frame version.                        |
+|`EINVAL_FRAME_TYPE`   |`0x100c`| Invalid IEEE 802.15.4 frame type.                           |
+|`EINVAL_ADDR_MODE`    |`0x100e`| Invalid IEEE 802.15.4 address mode.                         |
+|`EINVAL_SCF`          |`0x100f`| Invalid IEEE 802.15.4 security control field.               |
+|`EINVAL_FRAME`        |`0x1010`| Invalid IEEE 802.15.4 frame (generic).                      |
+|`EINVAL_CHAN_FIXED`   |`0x1011`| Invalid fixed channel.                                      |
+|`ENOTSUP`             |`0x2000`| Unsupported feature (generic).                              |
+|`ENOTSUP_FHSS_DEFAULT`|`0x2001`| Unsupported configuration mode for selected FHSS type ([`REQ_DATA_TX`][tx-req]).|
+
+### `0x06 SET_HOST_API`
+
+Inform the RCP about the host API version. If not send, the RCP will assume
+`2.0.0`. This command should be sent after `IND_RESET`, before any other
+command.
+
+ - `uint32_t api_version`  
+    Must be at least `0x02000000`.
+
+## Send and receive data
+
+Only a subset of the IEEE 802.15.4 frame formats are supported for the needs of
+Wi-SUN:
+
+  - Frame type must be `1` (data) or `2` (ack, [`CNF_DATA_TX`][tx-cnf] only).
+  - Frame version must be `2`.
+  - Source address mode must be `3`.
+  - Destination address mode must be either `0` or `3`.
+  - See ["Security"][sec] for limits on the auxiliary security header.
+
+[cca]:    #channel-access-and-retries
+[tx-req]: #0x10-req_data_tx
+[tx-cnf]: #0x12-cnf_data_tx
+[rx]:     #0x13-ind_data_rx
+
+### Channel Access and Retries
+
+In Wi-SUN, all transmissions require the use of Clear Channel Assessment (CCA),
+but the way attempts are performed differs depending on the [FHSS type][fhss]
+used.
+
+Unicast and broadcast to FFNs use the CSMA-CA algorithm with the following
+parameters:
+
+| IEEE 802.15.4 MAC PIB | Value | Description |
+|-----------------------|-------|-------------|
+| `macMinBe`            | `3`   | Minimum Backoff Exponent (BE) for CSMA-CA.
+| `macMaxBe`            | `5`   | Maximum Backoff Exponent (BE) for CSMA-CA.
+| `macMaxCsmaBackoffs`  | `8`   | Number of CSMA-CA backoffs to attempt before declaring failure.
+| `macMaxFrameRetries`  | `19`  | Number of retries allowed after a transmission failure.
+
+Unicast and broadcast transmissions to LFNs can be conceived as if they use
+the CSMA-CA algorithm, where the backoff is not randomized but instead
+corresponds to the next available TX slot. This detail matters for counting
+CCA failures versus transmission failures.
+
+Asynchronous transmissions use a CCA per channel, and skip the channel on
+failure. No overall CCA failure can be returned in [`CNF_DATA_TX`][tx-cnf]
+with these transmissions.
```

Comment:
Do we have to explain what `Asynchronous transmissions` are?

To be clear, I would insist the CSMA backoff is not used in asynchronous transmissions.
I would also say that `no retry is performed on failure` instead of `skip the channel on failure.`

Reply:
> Do we have to explain what `Asynchronous transmissions` are?

I don't think so, for the same reason that we should not explain CSMA-CA in details but instead only provide the parameters.